app/src/lib/menu-update.ts

```diff
@@ -260,6 +263,11 @@ function getRepositoryMenuBuilder(state: IAppState): MenuStateBuilder {
       !tipStateIsUnknown && !branchIsUnborn && !rebaseInProgress
     )
 
+    menuStateBuilder.setEnabled(
+      'discard-all-changes',
+      repositoryActive && hasChangedFiles && !rebaseInProgress
```

Comment:
Any other conditions we need to take into account? The previous logic (for the change list context menu) had:

```ts
// need to preserve the working directory state while dealing with conflicts
if (this.props.rebaseConflictState !== null) {
  return
}
```

Reply:
I think this is fine